### ISSUE TITLE
Issue with wrap in ui

### ISSUE THREAD

**Issue by chiru9670** [XI-E/LHospital#6]
The wrap function is messing up; this is the code:
```cpp
menu2 << "This program will exit now. You wanna go back, you're gonna have to start the program again." << ui::endl;
```
where menu2 is:
```cpp
box menu2(c, window.getwidth() * 0.6, window.getheight() - 6 );
menu2.clear();
menu2.settcolor(GREEN);
menu2 << ui::centeralign << "Employee Management" << ui::endl << ui::endl;
menu2.settcolor(WHITE);
```

The output is:
![capture](https://user-images.githubusercontent.com/25587869/34288169-9012d244-e711-11e7-9d6c-5627002c8ee7.JPG)

**Comment by chiru9670** [comment]
Am I doing something wrong, btw?

**Comment by arpit-saxena** [comment]
It looks alright to me. I'll have to check what's wrong with the wrap function

**Comment by arpit-saxena** [comment]
It does look like to me, tho, that you hit an edge case that I didn't find when I was testing he wrap function. See how "program" is just at the edge.

**Comment by arpit-saxena** [comment]
I fixed the issue and I'll close it when I commit my code.

Here is the code I use for the output shown after it:
```cpp
box menu2(coord(1,1),ui::scr_width * 0.6, ui::scr_height - 6 );
menu2.clear();
menu2.settcolor(GREEN);
menu2 << ui::centeralign << "Employee Management" << ui::endl << ui::endl;
menu2.settcolor(WHITE);

menu2 << "This program will exit now. You wanna go back, you're gonna have to start the program again." << ui::endl;
```

Output:
![screenshot from 2018-01-15 16-52-58](https://user-images.githubusercontent.com/14573995/34940232-c5ffc2ac-fa14-11e7-8c27-4a88d85bf60a.png)

**Comment by arpit-saxena** [comment]
Fixed as of https://github.com/XI-E/LHospital/commit/32ca022755ff968c3334874258f37663995191f0

**Comment by chiru9670** [comment]
Oi, the error is not fixed...
I think you made the box one character wider while testing it(I dunno why, your code is the same as mine...)
I used the same code as you did while testing it, i.e
```cpp
ui::clrscr();
box menu2(coord(1,1),ui::scr_width * 0.6, ui::scr_height - 6 );
menu2.clear();
menu2.settcolor(GREEN);
menu2 << ui::centeralign << "Employee Management" << ui::endl << ui::endl;
menu2.settcolor(WHITE);

menu2 << "This program will exit now. You wanna go back, you're gonna have to start the program again." << ui::endl;
```
and the output is:
![capture](https://user-images.githubusercontent.com/25587869/35059372-ebac8828-fbe0-11e7-8f1e-d752b278e433.JPG)

**Comment by arpit-saxena** [comment]
Yeah, it looks like I fixed another bug that day.
I had to modify the width of the box a bit to reproduce the error you were facing. This is the code I used:
```cpp
ui::clrscr();
box menu2(coord(1,1),47, ui::scr_height - 6 );
menu2.clear();
menu2.settcolor(GREEN);
menu2 << ui::centeralign << "Employee Management" << ui::endl << ui::endl;
menu2.settcolor(WHITE);

menu2 << "This program will exit now. You wanna go back, you're gonna have to start the program again.";
```
Output:
![screenshot from 2018-01-18 06-20-34](https://user-images.githubusercontent.com/14573995/35075211-cf9f813a-fc17-11e7-97df-52209be3591c.png)

I found the error, which was quite critical and led to wonder how the wrap was even working properly. Anyhow, here's the output after it was fixed:
![screenshot from 2018-01-18 06-16-02](https://user-images.githubusercontent.com/14573995/35075232-eb9d363e-fc17-11e7-9ea4-26b35fc8bb06.png)
